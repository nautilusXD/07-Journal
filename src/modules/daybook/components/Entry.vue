<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: 10 } })"
  >
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">28</span>
      <span class="mx-1 fs-5">Octubre</span>
      <span class="mx-2 fw-light">2021, Jueves</span>
    </div>
    <div class="entry-description">
      Cupidatat enim consequat qui cillum. Sunt tempor exercitation ullamco
      labore excepteur veniam velit dolore aliqua reprehenderit. Excepteur
      cupidatat commodo voluptate irure veniam aliquip proident. Eu consequat
      pariatur veniam quis aliqua elit incididunt irure do consequat deserunt
      ipsum incididunt proident. Sit proident minim id sint ipsum duis labore
      sunt ullamco voluptate proident commodo.
    </div>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }
  .entry-description {
    font-size: 12px;
  }
}
</style>
