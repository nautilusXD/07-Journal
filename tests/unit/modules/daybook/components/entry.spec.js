import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";
import { journalState } from "../../../mock-data/test-journal-state";

describe("Pruebas en Entry component", () => {
    const mockRouter = {
        push: jest.fn(),
    };

    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0],
        },
        global: {
            mocks: {
                $router: mockRouter,
            },
        },
    });

    test("debe de hacer match con el snapshot ", () => {
        expect(wrapper.html).toMatchSnapshot();
    });

    test("debe de redireccionar al hacer click en el entry-container ", () => {
        const entryContainer = wrapper.find(".entry-container");

        entryContainer.trigger("click");

        // expect(mockRouter.push).toHaveBeenCalled();

        expect(mockRouter.push).toHaveBeenCalledWith({
            name: "entry",
            params: {
                id: journalState.entries[0].id,
            },
        });
    });

    test("prueba en las propiedades computadas", () => {
        expect(wrapper.vm.day).toBe(29);
        expect(wrapper.vm.month).toBe("Octubre");
        expect(wrapper.vm.yearDay).toBe("2021, Viernes");
    });
});