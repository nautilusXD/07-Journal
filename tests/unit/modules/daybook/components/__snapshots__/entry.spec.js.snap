// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Entry component debe de hacer match con el snapshot  1`] = `[Function]`;
